### PR TITLE
feat: adhere to ruff's `RUF` rules

### PR DESCRIPTION
**Pull request by dni**
https://docs.astral.sh/ruff/rules/#ruff-specific-rules-ruf

#2308

### CODE REVIEW

File: lnbits/app.py [hunk 1]
```diff
@@ -262,7 +262,7 @@ async def check_installed_extension_files(ext: InstallableExtension) -> bool:
 
     zip_files = glob.glob(os.path.join(settings.lnbits_data_folder, "zips", "*.zip"))
 
-    if f"./{str(ext.zip_path)}" not in zip_files:
+    if f"./{ext.zip_path!s}" not in zip_files:
```
Comment (motorina0) [inline lnbits/app.py:265]: was ist das?

File: lnbits/core/models.py [hunk 2]
```diff
@@ -177,7 +177,7 @@ class Payment(FromRowModel):
     preimage: str
     payment_hash: str
     expiry: Optional[float]
-    extra: Dict = {}
+    extra: Dict
```
Comment (motorina0) [inline lnbits/core/models.py:180]: why not assign it a default value?

File: lnbits/core/models.py [hunk 3]
```diff
@@ -318,7 +318,7 @@ class PaymentFilters(FilterModel):
     preimage: str
     payment_hash: str
     expiry: Optional[datetime.datetime]
-    extra: Dict = {}
+    extra: ClassVar[Dict] = {}
```
Comment (motorina0) [inline lnbits/core/models.py:321]: what does `ClassVar` mean in this context?

File: lnbits/core/sso/keycloak.py [hunk 4]
```diff
@@ -11,7 +11,7 @@ class KeycloakSSO(SSOBase):
     """Class providing login via Keycloak OAuth"""
 
     provider = "keycloak"
-    scope = ["openid", "email", "profile"]
+    scope: list[str] = ["openid", "email", "profile"]  # noqa: RUF012
```
Comment (motorina0) [inline lnbits/core/sso/keycloak.py:14]: why no use `ClassaVar` here?

File: lnbits/db.py [hunk 5]
```diff
@@ -370,7 +370,7 @@ def from_row(cls, row: Row):
 
 
 class FilterModel(BaseModel):
-    __search_fields__: List[str] = []
+    __search_fields__: ClassVar[List[str]] = []
```
Comment (motorina0) [inline lnbits/db.py:373]: Aren't `__sort_fields__` below in the same situation?

File: lnbits/wallets/fake.py [hunk 6]
```diff
@@ -31,8 +31,8 @@
 
 class FakeWallet(Wallet):
     queue: asyncio.Queue = asyncio.Queue(0)
-    payment_secrets: Dict[str, str] = dict()
-    paid_invoices: Set[str] = set()
+    payment_secrets: ClassVar[Dict[str, str]] = {}
+    paid_invoices: ClassVar[Set[str]] = set()
```
Comment (motorina0) [inline lnbits/wallets/fake.py:35]: how was decided that only these two vars are `ClassVar` while the other not?

File: lnbits/app.py (hunk 1 above)
Comment (dni) [reply lnbits/app.py:265]: !s 	Interpolates the string representation from the .__str__() method
:D

File: lnbits/wallets/fake.py (hunk 6 above)
Comment (dni) [reply lnbits/wallets/fake.py:35]: ClassVar marks which are mutable object
https://docs.astral.sh/ruff/rules/mutable-class-default/

**Pull request closed without merging**